### ISSUE TITLE
[W4]-Twitter Client Assignment

### ISSUE THREAD

**Issue by calebripley** [calebripley/Twitter-Client#3]
My app is complete! /cc @codepathreview

**Comment by avo1** [comment]
Hi @calebripley ,

I don't see the last features implemented in either your gif, or your code. If you think this is a mistake, please **push your latest code asap** and reply to this issue to get my immediate attention. Do not create a new issue.

You created the retweet and fav API call, but nothing inside. You also didn't display the fav/retweet count.

<img width="660" alt="screen shot 2016-02-23 at 11 43 29 pm" src="https://cloud.githubusercontent.com/assets/10482065/13258889/96e5f3d2-da87-11e5-93ef-dc3f6190a054.png">

Dave
CodePath

**Comment by calebripley** [comment]
Hello Dave,

When you say "last features" are you referring to just the retweet and favorite features? I noticed that when I retweet or favorite, the retweet and favorite counts do increment on twitter, but I was not displaying the number on my app because it was not listed in the instructions on the code path website. Also, I am not sure what I am supposed to put inside of the API call, is that where I would increment the count on my local app? Thank you for the help.

Best,
- Caleb Ripley

**Comment by avo1** [comment]
Yes, I meant retweet and fav count. You can see find the retweet and favorite count in the json return by API and you should display that number. 

https://dev.twitter.com/rest/reference/get/statuses/home_timeline

![2](https://cloud.githubusercontent.com/assets/10482065/13278500/88275d5a-db03-11e5-87ea-c244cb3f7c85.PNG)

![capture1](https://cloud.githubusercontent.com/assets/10482065/13278507/8ddc9206-db03-11e5-9cd8-061a3adf33a5.PNG)

Inside your API call, you should check if your API call was successfully call, or error was returned.

**Comment by calebripley** [comment]
Hello Dave,

While working on this issue I keep running into this 
It seems using this method I am passing in a null value every time and the app crashes. How can I be sure that I am passing in the tweet that was favorited/ retweeted and not a nil value?
Thank you for the help.
- Caleb

**Comment by avo1** [comment]
Hi,

Sorry for late response. Not sure if you are still stuck at this point, but here's the hint. If you read in the API documentation, you see fav or retweet api requires the 1 param: id. You get the tweetID and pass it to the API call. Something likes this: TwitterClient.sharedInstance.likeStatus(["id" : tweetId], ......
